src/main.rs

```diff
@@ -535,79 +535,80 @@ impl ManParser for Type1 {
         manpage.contains(r#".SH "OPTIONS""#)
     }
 
-    fn parse_man_page(&mut self, manpage: &str) -> Option<String> {
+    fn parse_man_page(&mut self, manpage: &str, cmd_name: &str) -> Option<String> {
+        let options_section_re = regex!(r#"\.SH "OPTIONS"((?s:.)*?)(\.SH|\z)"#);
+        let options_section_matched = options_section_re.find(manpage);
+
+        let mut options_section = options_section_matched.unwrap().as_str();
+        let options_parts_re = regex!(r"\.PP((?s:.)*?)\.RE");
+        let mut options_matched = options_parts_re.captures(options_section);
+        // add_diagnostic(format!("Command is {}", cmd_name));
+
+        if options_matched.is_none() {
+            // add_diagnostic("Unable to find options");
+            return self
+                .fallback(options_section, cmd_name)
+                .or_else(|| self.fallback2(options_section, cmd_name));
+        }
+
+        let mut built_command_output = Vec::new();
+        let mut existing_options = HashSet::new();
+        while let Some(mat) = options_matched {
+            let mut data = mat.get(1).unwrap().as_str();
+            let last_dotpp_index = data.rfind(".PP");
+            if let Some(idx) = last_dotpp_index {
+                data = &data[idx + 3..];
+            }
+
+            let data = remove_groff_formatting(data);
+            let data: Vec<_> = data.split(".RS 4").collect();
+            if data.len() > 1 {
+                let option_name = data[0].trim();
+                if option_name.find('-').is_some() {
+                    let option_name = unquote_double_quotes(option_name);
+                    let option_name = unquote_single_quotes(option_name);
+                    let option_desc = data[1].trim().replace('\n', " ");
+                    built_command(
+                        option_name,
+                        option_desc.as_str(),
+                        &mut built_command_output,
+                        &mut existing_options,
+                        cmd_name,
+                    );
+                } else {
+                    // add_diagnostic(format!("{:?} doesn't contain '-'", option_name));
```

Comment:
I think we need to decide how functions will have access to an `App` and its methods. Thanks for putting these in here now though

Reply:
I think using `log` would be better for this. I talked to @scooter-dangle about it, looks like two popular crate that we need to use are either `simple_logger` and `env_logger` (which I prefer).